src/cursor/cur_table.c

```diff
@@ -837,18 +837,39 @@ __curtable_largest_key(WT_CURSOR *cursor)
 static int
 __curtable_bound(WT_CURSOR *cursor, const char *config)
 {
-    WT_CURSOR *primary;
+    WT_CURSOR **cp, *primary;
+    WT_CURSOR_BOUNDS_STATE saved_bounds;
     WT_CURSOR_TABLE *ctable;
     WT_DECL_RET;
     WT_SESSION_IMPL *session;
+    u_int i;
 
     ctable = (WT_CURSOR_TABLE *)cursor;
+    primary = *ctable->cg_cursors;
     JOINABLE_CURSOR_API_CALL(cursor, session, bound, NULL);
 
-    /* Grab the primary cursor and call bound function. */
-    primary = *ctable->cg_cursors;
-    WT_ERR(primary->bound(primary, config));
+    saved_bounds.lower_bound = NULL;
+    saved_bounds.upper_bound = NULL;
+
+    /* Save the current state of the bounds in case we fail to apply the new state. */
+    if (WT_CURSOR_BOUNDS_SET(primary)) {
```

Comment:
Given that we check both flags inside this call I think we can remove this if statement.